massa-final-state/src/final_state.rs

```diff
@@ -240,8 +240,10 @@ impl FinalState {
                             .iter()
                             .filter_map(|change| match change {
                                 Change::Add(id, _) if id <= &last_id => Some(change.clone()),
+                                Change::Activate(id) if id <= &last_id => Some(change.clone()),
                                 Change::Delete(id) if id <= &last_id => Some(change.clone()),
                                 Change::Add(..) => None,
+                                Change::Activate(..) => None,
                                 Change::Delete(..) => None,
                             })
```

Comment:
This is way too verbose for nothing, please use:

```suggestion
                            .filter_map(|change| match change {
                                Change::Add(id, _) | Change::Activate(id) | Change::Delete(id)
                                    if id <= &last_id =>
                                {
                                    Some(change.clone())
                                }
                                _ => None,
                            })
```